pipeline_live/data/sources/alpaca.py

```diff
@@ -25,14 +26,20 @@ def _get_stockprices(symbols, limit=365, timespan='day'):
     '''
 
     api = tradeapi.REST()
+    timeframe = TimeFrame.Minute if timespan == "minute" else TimeFrame.Day
 
     def fetch(symbols):
-        barset = api.get_barset(symbols, timespan, limit)
         data = {}
-        for symbol in barset:
-            df = barset[symbol].df
+        for symbol in symbols:
+            df = api.get_bars(symbol,
+                              limit=limit,
+                              timeframe=timeframe,
+                              adjustment='raw').df
+
             # Update the index format for comparison with the trading calendar
-            df.index = df.index.tz_convert('UTC').normalize()
+            if not df.empty and df.index.tzinfo is None:
+                df.index = df.index.tz_localize('UTC')
+
             data[symbol] = df.asfreq('C')
```

Comment:
Is this frequency conversion to handle minute timeframe data specifically?

Reply:
Hmm the actual nature of how this gets called from the rest of the code it always is day but I didn't try with minute, I'll test and see :thinking:

Reply:
you're right, if it would somehow be minute it gets truncated down back into days